### ISSUE TITLE
Missing licence

### ISSUE THREAD

**Issue by laundmo** [StressLevelZero/MarrowSDK#1]
re: https://github.com/StressLevelZero/MarrowSDK/wiki/License

Without a licence, nobody is allowed to use this. See: https://choosealicense.com/no-permission/

Could you please add a licence to this?

**Comment by EsserGaming** [comment]
They have a [License Page](https://github.com/StressLevelZero/MarrowSDK/wiki/License) in their Wiki, but it doesn't seem to have anything yet.

**Comment by laundmo** [comment]
> They have a [License Page](https://github.com/StressLevelZero/MarrowSDK/wiki/License) in their Wiki, but it doesn't seem to have anything yet.

Yes. that's the first link in the issue text...

**Comment by oscar1000108** [comment]
This also means that any contributors from before a license is added will need to give explicit permission for their contributions to be made available under the new license.

This needs to be dealt with as a matter of urgency. Please choose wisely SLZ, as a bad license could kill the modding community before it even walks, or cause people to pursue creating third party SDKs.

**Comment by laundmo** [comment]
> This also means that any contributors from before a license is added will need to give explicit permission for their contributions to be made available under the new license.

To be fair, i doubt there will be many 3rd party contributions accepted at least in the first few weeks/months

BUT: more imporantly, even *using* the SDK is a issue as you have not been given permission to use it by a license.

**Comment by CuteWaterbeary** [comment]
Someone initiate a pull request with a good license cc0, mit, etc

**Comment by laundmo** [comment]
They're looking into it:
![image](https://user-images.githubusercontent.com/24855949/193378934-5ae04807-cc1f-4543-9209-b17d4dd44ecc.png)

**Comment by CuteWaterbeary** [comment]
How many years can you look into it for?

**Comment by TheRektafire** [comment]
> How many years can you look into it for? This company is no newcomer to game releases so I'm pretty sure this is a strategy from their legal dept.

Yeah this is honestly kind of a joke IMO especially coming from a team as at least superficially seemingly competent as SLZ, you'd think they would have figured out what the license would be way before release, but instead not only did they apparently not think of that but they went out of their way to straight up release it without including one??? It's basically unusable right now as a result and considering they probably want a modding community to form as quickly as possible, especially considering the mixed reviews and opinions on the base game at the moment, they need to sort this out pretty quick. Hopefully they go with a nice permissive one like MIT or at the very least GPL like doom/quake...

**Comment by laundmo** [comment]
i doubt they'll want to go with a fully Open Source license, a source-available one which specifies the usage terms for the SDK seems much more likely, but i hope to be surprised

**Comment by CuteWaterbeary** [comment]
The EULA for the game makes it seem like an open source license might be the last thing to happen. My fav parts:

`You agree to indemnify, defend and hold harmless Stress Level Zero INC, its partners, affiliates, contractors, licensors, officers, directors, employees and agents from all claims, damages, costs and expenses (including reasonable legal fees) arising directly or indirectly from your acts or omissions in connection with using the Game Software or any breach by you of the terms of this Agreement.`

`YOU ACKNOWLEDGE THAT YOU ARE VOLUNTARILY PARTICIPATING IN THE USE OF THE GAME SOFTWARE ENTIRELY AT YOUR OWN RISK. YOU ARE AWARE OF THE RISKS ASSOCIATED WITH USING THE GAME SOFTWARE, WHICH MAY INCLUDE, BUT ARE NOT LIMITED TO PHYSICAL OR PSYCHOLOGICAL INJURY, PAIN, SUFFERING, ILLNESS, DISFIGUREMENT, TEMPORARY OR PERMANENT DISABILITY (INCLUDING PARALYSIS), ECONOMIC OR EMOTIONAL LOSS, AND DEATH. `

**Comment by dotbenj** [comment]
You guys are going pretty heavy on them. I'm following early node and SLZ since their very begining and I haven't heard of any big legal issues on any of Boneworks modding.

The SDK is still an hatching eggs (0.3.5) as I understand thing they were more worry about releasing something players can experiments than something lawers can and to me it's a good point.

I maybe wrong and future will correct me if I am but if they want a strong open source community growing around their game they'll probably come with a licence that will be open.

Speaking about EULA it's a pretty standard thing you need to come up in a country where you could be sued for any stupid thing someone posted on YouTube about your game ...

**Comment by dotbenj** [comment]
Should close the discussion:
bc50d82a1176d964d8f1f65e976e9fd2ef5841c1

**Comment by CuteWaterbeary** [comment]
> Should close the discussion: [bc50d82](https://github.com/StressLevelZero/MarrowSDK/commit/bc50d82a1176d964d8f1f65e976e9fd2ef5841c1)

for adding the license from WebSocketSharp covering only WebSocketSharp?

**Comment by laundmo** [comment]
> Should close the discussion: [bc50d82](https://github.com/StressLevelZero/MarrowSDK/commit/bc50d82a1176d964d8f1f65e976e9fd2ef5841c1)

That's not what this issue is about, the SDK itself is still missing the license which is the subject of this issue

**Comment by CuteWaterbeary** [comment]
Please keep the issue open until it is resolved.

**Comment by laundmo** [comment]
Seconded, please reopen the issue. Issues should only be closed if they're resolved or won't be resolved.

**Comment by yewnyx** [comment]
Apologies for jumping the  gun a bit. We've decided to move forward with the MIT license.

As mentioned in one of the other issues, this repository is generated through our CI. We're in the process of updating this repository - sit tight.

**Comment by laundmo** [comment]
Wow, thank you. i was not expecting a Open Source license!

> As mentioned in one of the other issues, this repository is generated through our CI. We're in the process of updating this repository - sit tight.

This makes me wonder whether you will accept contributions and how that will work with the CI generating this Repo.

**Comment by CuteWaterbeary** [comment]
Thank you everyone for resolving this and with a wonderful license too.
